packages/subgraph/src/mappings/account-handler.ts

```diff
@@ -535,3 +577,83 @@ export function handleAllowedSellersRemovedEvent(
     disputeResolverId.toString()
   );
 }
+
+export function getOfferCollectionId(
+  sellerId: string,
+  collectionIndex: string
+): string {
+  const offerCollectionId = sellerId + "-collection-" + collectionIndex;
+  return offerCollectionId;
+}
+
+function saveOfferCollection(
+  offerCollectionId: string,
+  sellerId: BigInt,
+  collectionIndex: BigInt,
+  collectionAddress: Address,
+  externalIdHash: Bytes,
+  externalId: string
+): void {
+  let offerCollection = OfferCollection.load(offerCollectionId);
+
+  if (!offerCollection) {
+    offerCollection = new OfferCollection(offerCollectionId);
+    offerCollection.sellerId = sellerId;
+    offerCollection.seller = sellerId.toString();
+    offerCollection.collectionIndex = collectionIndex;
+    offerCollection.collectionAddress = collectionAddress;
+    offerCollection.externalIdHash = externalIdHash;
+    offerCollection.externalId = externalId;
+    offerCollection.metadata = offerCollectionId;
+    offerCollection.save();
+  } else {
+    log.warning("Offer collection with ID '{}' already exists!", [
+      offerCollectionId
+    ]);
+  }
```

Comment:
NIT: swap conditions

```suggestion
  if (offerCollection) {
     log.warning("Offer collection with ID '{}' already exists!", [
        offerCollectionId
      ]);
  } else {
    offerCollection = new OfferCollection(offerCollectionId);
    offerCollection.sellerId = sellerId;
    offerCollection.seller = sellerId.toString();
    offerCollection.collectionIndex = collectionIndex;
    offerCollection.collectionAddress = collectionAddress;
    offerCollection.externalIdHash = externalIdHash;
    offerCollection.externalId = externalId;
    offerCollection.metadata = offerCollectionId;
    offerCollection.save();
  }
```